### DISCUSSION TITLE
HTTP3/QUIC Support

### DISCUSSION THREAD

**Discussion by deepal** [Kong/insomnia#3691] [Ideas]
**Is your feature request related to a problem? Please describe.**
The latest HTTP version Insomnia supports is HTTP/2 and it would be amazing to see HTTP3 support too, now that it's widely available via many CDNs, supported by Chrome by default, and being rolled out by Firefox  at the time of the writing. [Curl also has experimental support for HTTP3](https://github.com/curl/curl/blob/master/docs/HTTP3.md) if built from source. 

Not sure about the feasibility at the moment and if it's too early to ask for HTTP3 in insomnia, but nevertheless logging this issue since I didn't find any issue related to this.

**Describe the solution you'd like**
Use HTTP3 for requests if the server supports it.

**Describe alternatives you've considered**
There are not many Rest/HTTP clients out there that supports HTTP3, but `curl` has [experimental support](https://github.com/curl/curl/blob/master/docs/HTTP3.md) if built from source specifically.

**Additional context**
NodeJS support: https://github.com/nodejs/node/issues/38478
Original work for NodeJS: https://github.com/nodejs/quic/blob/master/doc/api/quic.md

**Comment by dimitropoulos** [comment]
does node-libcurl have (experimental) support for this?  I ask because I'm not sure how current what they build is with respect to curl itself.
